File: api/src/main/java/com/delivery/api/common/status/SuccessStatusCode.java
package com.delivery.api.common.status;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@Getter
@RequiredArgsConstructor
public enum SuccessStatusCode implements StatusCode {
    OK(HttpStatus.OK.value(), 200, "성공");

    private final Integer httpStatusCode;
    private final Integer code;
    private final String message;
}

File: common/src/main/kotlin/com/delivery/common/status/SuccessStatusCode.kt
package com.delivery.common.status

import org.springframework.http.HttpStatus

enum class SuccessStatusCode(
    private val httpStatusCode: Int,
    private val code: Int,
    private val message: String,
) : StatusCode {
    OK(HttpStatus.OK.value(), 200, "성공")
    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getCode(): Int {
        return this.code
    }

    override fun getMessage(): String {
        return this.message
    }
}